.github/workflows/mango_base_unit_testing.yml

```diff
@@ -0,0 +1,88 @@
+name: mango base full unit testing
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    paths:
+      - 'mango/pyproject.toml'
+      - 'mango/mango/**'
+      - '.github/workflows/**base**'
+  push:
+    branches:
+      - master
+      - develop
+
+jobs:
+  unit-testing:
+    name: Run all unit test suites
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 21
+      matrix:
+        python-version: [3.8, 3.9, '3.10', '3.11', '3.12']
+        os: [ubuntu-latest, windows-latest, macos-latest]
+
+    steps:
+      - uses: actions/checkout@v1
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: 'pip'
+          cache-dependency-path: './requirements-dev.txt'
+      - name: Install macOS dependencies
+        if: matrix.os == 'macos-latest'
```

Comment:
If we add macos-latest in mango_all_unit_testing.yml we should add this line too (Dependency for LightGBM)